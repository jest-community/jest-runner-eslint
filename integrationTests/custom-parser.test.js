const eslint = require('eslint');
const runJest = require('./runJest');

// Note: ESLint versions <6 have a different error message for this test. The
// snapshot file contains both messages so we can test across both versions.
// Without the skipped tests for the "other" version, the tests will always fail
// with `1 snapshot obsolete`.
if (
  eslint.CLIEngine.version.startsWith('4') ||
  eslint.CLIEngine.version.startsWith('5')
) {
  it.skip("Doesn't override parser when not set", () => {});
  it("Doesn't override parser when not set [ESLint<6]", async () => {
    expect(await runJest('custom-parser')).toMatchSnapshot();
  });
} else {
  it("Doesn't override parser when not set", async () => {
    expect(await runJest('custom-parser')).toMatchSnapshot();
  });
  it.skip("Doesn't override parser when not set [ESLint<6]", () => {});
}
