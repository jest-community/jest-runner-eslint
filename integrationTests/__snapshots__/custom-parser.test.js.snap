// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doesn't override parser when not set [ESLint=7] 1`] = `
"FAIL __eslint__/file.js
  ● Test suite failed to run

    Failed to load parser 'this-parser-does-not-exist' declared in '.eslintrc.json': Cannot find module 'this-parser-does-not-exist'
    Require stack:
    - /mocked-path-to-jest-runner-mocha/integrationTests/__fixtures__/custom-parser/.eslintrc.json

      at Object.resolve (../../../node_modules/@eslint/eslintrc/lib/shared/relative-module-resolver.js:28:50)
          at _normalizeObjectConfigDataBody.next (<anonymous>)
          at _normalizeObjectConfigData.next (<anonymous>)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        
Ran all test suites.
"
`;

exports[`Doesn't override parser when not set 1`] = `
"FAIL __eslint__/file.js
  ● Test suite failed to run

    Failed to load parser 'this-parser-does-not-exist' declared in '.eslintrc.json': Cannot find module 'this-parser-does-not-exist'
    Require stack:
    - /mocked-path-to-jest-runner-mocha/integrationTests/__fixtures__/custom-parser/.eslintrc.json

      at Object.resolve (../../../node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2337:46)
          at _normalizeObjectConfigDataBody.next (<anonymous>)
          at _normalizeObjectConfigData.next (<anonymous>)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        
Ran all test suites.
"
`;
